models.py

```diff
@@ -15,104 +15,134 @@ class EVC:
 
-        Raises:
-            TypeError: Rases an error if the error message.
-        """
-        for attribute in self.required_attributes:
-            if attribute not in kwargs.keys():
-                raise TypeError(f"{attribute} is required.")
-            if 'uni' in attribute:
-                uni = kwargs.get(attribute)
-                if not isinstance(uni, UNI):
-                    raise TypeError(f"Invalid UNI: {attribute}.")
-                if not uni.is_valid():
-                    raise TypeError("Invalid UNI {attribute}.")
+        if not uni_z.is_valid():
+            raise TypeError("Invalid uni_z.")
 
     def as_dict(self):
-        """Dict representation for the EVC object."""
-        evc_dict = {}
-
-        for attribute in self.default_attributes:
-            if '_id' in attribute:
-                evc_dict['id'] = getattr(self, attribute)
-            elif 'time' in attribute or 'date' in attribute:
-                value = getattr(self, attribute)
-                if value:
-                    evc_dict[attribute] = value.strftime("%Y-%m-%dT%H:%M:%S")
-            elif 'uni' in attribute:
-                uni = getattr(self, attribute)
-                evc_dict[attribute] = uni.as_dict()
-            else:
-                evc_dict[attribute] = getattr(self, attribute)
+        """A dictionary representing an EVC object."""
+        evc_dict = {"id": self.id, "name": self.name,
+                    "uni_a": self.uni_a.as_dict(),
+                    "uni_z": self.uni_z.as_dict()}
+
+        if self.start_date:
+            date = self.start_date.strftime("%Y-%m-%dT%H:%M:%S")
+            evc_dict["start_date"] = date
+
+        if self.end_date:
+            date = self.end_date.strftime("%Y-%m-%dT%H:%M:%S")
+            evc_dict['end_date'] = date
+
+        if self.bandwidth:
+            evc_dict['bandwidth'] = self.bandwidth
+
+        if self.primary_links:
+            evc_dict['primary_links'] = self.primary_links
+
+        if self.backup_links:
+            evc_dict['backup_links'] = self.backup_links
+
+        if self.primary_links:
+            evc_dict['dynamic_backup_path'] = self.dynamic_backup_path
+
+        if self._requested:
+            evc_dict['_requested'] = self._requested
+
+        if self.current_path:
+           evc_dict['current_path'] = self.current_path
+
+        if self.primary_path:
+           evc_dict['primary_path'] = self.primary_path
+
+        if self.backup_path:
+           evc_dict['backup_path'] = self.backup_path
+
+        if self.request_time:
+           time = self.request_time.strftime("%Y-%m-%dT%H:%M:%S")
+           evc_dict['request_time'] = time
+
+        if self.owner:
+           evc_dict['owner'] = self.owner
+
+        if self.active:
+           evc_dict['active'] = self.active
+
+        if self.enabled:
+           evc_dict['enabled'] = self.enabled
+
+        if self.priority:
+           evc_dict['priority'] = self.priority
+
```

Comment:
Does the blueprint specify if attributes that are not set should not be returned in the dict, or if they should always be returned with an empty string/null value?